QModManager/API/ModLoading/Internal/PatchMethod.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace QModManager.API.ModLoading.Internal
+{
+    using System;
+    using System.Reflection;
+    using QModManager.Utility;
+
+    internal class PatchMethod
+    {
+        public PatchMethod(MethodInfo method, IQModLoadable qmod)
+        {
+            this.ModId = qmod.Id;
+            this.Method = method;
+            this.LoadedAssembly = qmod.LoadedAssembly;
+        }
+
+        internal string ModId { get; }
+        internal MethodInfo Method { get; }
+        internal Assembly LoadedAssembly { get; }
+        internal bool IsPatched { get; private set; }
+
+        internal bool TryInvoke()
+        {
+            try
+            {
+                this.Method.Invoke(this.LoadedAssembly, new object[] { });
```

Comment:
These methods _shouldn't_ be requiring parameters in the first place.  
But yes, we can check this for safety to avoid an unnecessary exception catch.

Reply:
I think that this should be done in the attribute class itself.

Reply:
Are you suggesting we merge the functionality of `QModPatchAttributeBase` and `QModPatchMethod` into a single class?  
I think it's doable. I'm on the fence about it right now but I don't have any strong reasons not to do it if you think it'll make sense to make it happen.